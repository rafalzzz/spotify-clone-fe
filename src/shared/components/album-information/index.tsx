import { useRouter } from 'next/navigation';
import { FC } from 'react';

import { generateArtistRedirectionPath } from '@/utils/generate-artist-redirection-path';

import { TAlbumInformation } from '@/types/components';

import { CustomTooltip } from '../custom-tooltip';

import './AlbumInformation.scss';

export const AlbumInformation: FC<TAlbumInformation> = ({
  collectionName,
  releaseDate,
  artistId,
  artistName,
}): JSX.Element => {
  const router = useRouter();
  const releaseYear = releaseDate.split('-')[0];

  const handleArtistClick = () => {
    const path = generateArtistRedirectionPath(artistId);
    router.push(path);
  };

  return (
    <div className='album-information'>
      <CustomTooltip title={collectionName} testId='album-information-collection-name-tooltip'>
        <h3 className='album-information__text' data-testid='album-information-collection-name'>
          {collectionName}
        </h3>
      </CustomTooltip>
      <div className='album-information__wrapper'>
        <span
          className='album-information__release-date'
          data-testid='album-information-release-date'
        >
          {releaseYear}
        </span>
        <span className='album-information__separator'>&#x2022;</span>
        <button
          className='album-information__text album-information__artist'
          data-testid='album-information-artist-name'
          onClick={handleArtistClick}
        >
          <CustomTooltip title={artistName} testId='album-information-artist-name-tooltip'>
            <span>{artistName}</span>
          </CustomTooltip>
        </button>
      </div>
    </div>
  );
};
