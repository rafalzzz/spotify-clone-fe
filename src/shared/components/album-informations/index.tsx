import { useRouter } from 'next/navigation';

import { generateArtistRedirectionPath } from '@/utils/generate-artist-redirection-path';

import { CustomTooltip } from '../custom-tooltip';

import './AlbumInformations.scss';

type AlbumInformationsProps = {
  collectionName: string;
  releaseDate: string;
  artistName: string;
};

export const AlbumInformations = ({
  collectionName,
  releaseDate,
  artistName,
}: AlbumInformationsProps) => {
  const router = useRouter();
  const releaseYear = releaseDate.split('-')[0];

  const handleArtistClick = () => {
    const path = generateArtistRedirectionPath(artistName);
    router.push(path);
  };

  return (
    <div className='album-informations'>
      <CustomTooltip title={collectionName} testId='album-informations-collection-name-tooltip'>
        <h4 className='album-informations__text' data-testid='album-informations-collection-name'>
          {collectionName}
        </h4>
      </CustomTooltip>
      <div className='album-informations__wrapper'>
        <span
          className='album-informations__release-date'
          data-testid='album-informations-release-date'
        >
          {releaseYear}
        </span>
        <span className='album-informations__separator'>&#x2022;</span>
        <button
          className='album-informations__text album-informations__artist'
          data-testid='album-informations-artist-name'
          onClick={handleArtistClick}
        >
          <CustomTooltip title={artistName} testId='album-informations-artist-name-tooltip'>
            <span>{artistName}</span>
          </CustomTooltip>
        </button>
      </div>
    </div>
  );
};
