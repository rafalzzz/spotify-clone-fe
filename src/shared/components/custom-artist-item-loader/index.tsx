import './CustomArtistItemLoader.scss';

export const CustomArtistItemLoader = () => (
  <div className='custom-artist-item-loader'>
    <div className='custom-artist-item-loader__skeleton-text'></div>
  </div>
);
