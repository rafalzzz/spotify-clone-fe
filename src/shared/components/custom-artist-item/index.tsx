import Link from 'next/link';

import { generateArtistRedirectionPath } from '@/utils/generate-artist-redirection-path';

import { Artist } from '@/shared/interfaces/artist';

import './CustomSectionItem.scss';

type TCustomArtistSection = Pick<Artist, 'artistName'>;

export const CustomArtistItem = ({ artistName }: TCustomArtistSection) => (
  <Link
    href={generateArtistRedirectionPath(artistName)}
    className='custom-artist-item'
    data-testid='custom-artist-item-redirection'
  >
    <div className='custom-artist-item__wrapper'>
      <div>{artistName}</div>
    </div>
  </Link>
);
