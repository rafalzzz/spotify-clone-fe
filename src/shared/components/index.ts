export { AlbumInformation } from './album-information';
export { CustomAddToFavoriteButton } from './custom-add-to-favorite-button';
export { CustomArtistItem } from './custom-artist-item';
export { CustomButton } from './custom-button';
export { CustomFormItem } from './custom-form-item';
export { CustomHeader } from './custom-header';
export { CustomHeadSection } from './custom-head-section';
export { CustomHyperlink } from './custom-hyperlink';
export { CustomMainContentWrapper } from './custom-main-content-wrapper';
export { CustomPageWrapper } from './custom-page-wrapper';
export { CustomSection } from './custom-section';
export { CustomSectionItem } from './custom-section-item';
export { CustomSectionItemImage } from './custom-section-item-image';
export { CustomSubheader } from './custom-subheader';
export { CustomTooltip } from './custom-tooltip';
export { MusicTrackInformation } from './music-track-informations';
export { SubmitButton } from './submit-button';
