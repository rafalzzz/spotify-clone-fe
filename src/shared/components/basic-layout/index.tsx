import { ReactNode } from 'react';

import { Sidebar } from '@/sidebar/components/sidebar';

import { Navbar } from '@/navigation/components/navbar';

import { CustomPageWrapper } from '@/components/custom-page-wrapper';

import './BasicLayout.scss';

type BasicLayoutProps = {
  children: ReactNode;
};

const BasicLayout = ({ children }: BasicLayoutProps) => (
  <CustomPageWrapper className='justify-start'>
    <Sidebar />
    <div className='basic-layout__content'>
      <Navbar />
      {children}
    </div>
  </CustomPageWrapper>
);

export default BasicLayout;
