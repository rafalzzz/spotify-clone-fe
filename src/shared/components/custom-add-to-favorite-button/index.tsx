import { HeartFilled } from '@ant-design/icons';
import { Button } from 'antd';
import React, { FC } from 'react';

import { TCustomAddToFavoriteButton } from '@/types/components';

import { CustomTooltip } from '../custom-tooltip';

import './CustomAddToFavoriteButton.scss';

export const CustomAddToFavoriteButton: FC<TCustomAddToFavoriteButton> = ({
  title = '',
  disabled = false,
  isAddedToFav = true,
  onClick,
}): JSX.Element => (
  <Button
    className={`custom-add-to-favorite-button custom-add-to-favorite-button--${
      isAddedToFav ? 'inactive' : 'active'
    }`}
    size='small'
    htmlType='button'
    onClick={onClick}
    role='button'
    data-testid='custom-add-to-favorite-button'
    disabled={disabled}
    ghost
  >
    <CustomTooltip title={title} placement='top' testId='custom-add-to-favorite-button-tooltip'>
      <HeartFilled />
    </CustomTooltip>
  </Button>
);
