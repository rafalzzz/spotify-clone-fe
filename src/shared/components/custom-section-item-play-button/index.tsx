import { PlayCircleFilled, PauseCircleFilled } from '@ant-design/icons';
import { FC } from 'react';

import { TCustomSectionItemPlayButton } from '@/types/components';

import './CustomSectionItemPlayButton.scss';

export const CustomSectionItemPlayButton: FC<TCustomSectionItemPlayButton> = ({
  isActive,
  isPlaying,
  onClick,
}): JSX.Element => {
  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    event.preventDefault();
    event.stopPropagation();
    onClick();
  };

  return (
    <button
      className={`custom-section-item-play-button ${
        isActive ? `custom-section-item-play-button--visible` : ''
      }`}
      onClick={handleClick}
      data-testid='custom-section-item-play-button'
    >
      {isPlaying && isActive ? (
        <PauseCircleFilled className='antd-icon' data-testid='custom-section-item-pause-icon' />
      ) : (
        <PlayCircleFilled className='antd-icon' data-testid='custom-section-item-play-icon' />
      )}
    </button>
  );
};
