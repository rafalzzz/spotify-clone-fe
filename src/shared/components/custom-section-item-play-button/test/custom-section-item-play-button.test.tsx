import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import { TCustomSectionItemPlayButton } from '@/types/components';

import { CustomSectionItemPlayButton } from '..';

const mockOnClick = jest.fn();

const PLAY_BUTTON_TEST_ID = 'custom-section-item-play-button';
const PLAY_ICON_TEST_ID = 'custom-section-item-play-icon';
const PAUSE_ICON_TEST_ID = 'custom-section-item-pause-icon';

const renderCustomSectionItemPlayButton = ({
  isActive = false,
  isPlaying = false,
}: Partial<TCustomSectionItemPlayButton> = {}) => {
  return render(
    <CustomSectionItemPlayButton isActive={isActive} isPlaying={isPlaying} onClick={mockOnClick} />,
  );
};

describe('CustomSectionItemPlayButton', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders component without error', () => {
    const screen = renderCustomSectionItemPlayButton();
    expect(screen).toMatchSnapshot();
  });

  it('calls onClick when the button is clicked', () => {
    const { queryByTestId } = renderCustomSectionItemPlayButton();

    const button = queryByTestId(PLAY_BUTTON_TEST_ID);
    fireEvent.click(button as Element);

    expect(mockOnClick).toHaveBeenCalled();
  });

  it('displays the play button with correct class when isActive is true', () => {
    const { queryByTestId } = renderCustomSectionItemPlayButton({ isActive: true });

    const button = queryByTestId(PLAY_BUTTON_TEST_ID);
    expect(button).toHaveClass('custom-section-item-play-button--visible');
  });

  it('displays the PlayCircleFilled icon when isPlaying is equal false', () => {
    const { queryByTestId } = renderCustomSectionItemPlayButton();

    const playIcon = queryByTestId(PLAY_ICON_TEST_ID);
    expect(playIcon).toBeInTheDocument();
  });

  it('displays the PauseCircleFilled icon when isPlaying and isActive are equal true', () => {
    const { queryByTestId } = renderCustomSectionItemPlayButton({
      isPlaying: true,
      isActive: true,
    });

    const pauseIcon = queryByTestId(PAUSE_ICON_TEST_ID);
    expect(pauseIcon).toBeInTheDocument();
  });
});
