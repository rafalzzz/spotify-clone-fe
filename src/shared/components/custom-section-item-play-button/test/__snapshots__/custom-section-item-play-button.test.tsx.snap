// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomSectionItemPlayButton renders component without error 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="custom-section-item-play-button "
        data-testid="custom-section-item-play-button"
      >
        <span
          aria-label="play-circle"
          class="anticon anticon-play-circle antd-icon"
          data-testid="custom-section-item-play-icon"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="play-circle"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm144.1 454.9L437.7 677.8a8.02 8.02 0 01-12.7-6.5V353.7a8 8 0 0112.7-6.5L656.1 506a7.9 7.9 0 010 12.9z"
            />
          </svg>
        </span>
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="custom-section-item-play-button "
      data-testid="custom-section-item-play-button"
    >
      <span
        aria-label="play-circle"
        class="anticon anticon-play-circle antd-icon"
        data-testid="custom-section-item-play-icon"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="play-circle"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm144.1 454.9L437.7 677.8a8.02 8.02 0 01-12.7-6.5V353.7a8 8 0 0112.7-6.5L656.1 506a7.9 7.9 0 010 12.9z"
          />
        </svg>
      </span>
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
